File: app/src/main/java/digispark/tech/ganapatiaarti/firebase/MyFirebaseMsgService.java
package digispark.tech.ganapatiaarti.firebase;

import android.util.Log;

import com.google.firebase.messaging.FirebaseMessagingService;
import com.google.firebase.messaging.RemoteMessage;

public class MyFirebaseMsgService extends FirebaseMessagingService {

    @Override
    public void onNewToken(String token) {
        Log.d("token", "Refreshed token: " + token);
    }

    @Override
    public void onMessageReceived(RemoteMessage remoteMessage) {
        super.onMessageReceived(remoteMessage);

        if (remoteMessage.getNotification() != null) {
            Log.d("onMessageReceived", "Message Notification Body: " + remoteMessage.getNotification().getBody());
            if (remoteMessage.getNotification() != null){
                String notiTitle = remoteMessage.getNotification().getTitle();
                String notiText = remoteMessage.getNotification().getBody();
                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O){
                    NotificationHelper.createNotificationChannel(this);
                }
                NotificationHelper.showNotification(this, notiTitle, notiText);
            }
        }
    }
}

File: app/src/main/java/digispark/tech/ganapatiaarti/firebase/MyFirebaseMsgService.kt
package digispark.tech.ganapatiaarti.firebase

import android.util.Log

import com.google.firebase.messaging.FirebaseMessagingService
import com.google.firebase.messaging.RemoteMessage

class MyFirebaseMsgService : FirebaseMessagingService() {

    override fun onNewToken(token: String) {
        Log.d("token", "Refreshed token: " + token)
    }

    override fun onMessageReceived(remoteMessage: RemoteMessage) {
        super.onMessageReceived(remoteMessage)

        if (remoteMessage.notification != null) {
            Log.d("onMessageReceived", "Message Notification Body: " + remoteMessage.notification!!.body!!)
            if (remoteMessage.notification != null) {
                val notiTitle = remoteMessage.notification!!.title
                val notiText = remoteMessage.notification!!.body
                if (android.os.Build.VERSION.SDK_INT >= android.os.Build.VERSION_CODES.O) {
                    NotificationHelper.createNotificationChannel(this)
                }
                NotificationHelper.showNotification(this, notiTitle!!, notiText!!)
            }
        }
    }
}
